crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -196,12 +203,25 @@ where
             state_root_config.prefix_sets.clone(),
         );
         let max_proof_task_concurrency = config.max_proof_task_concurrency() as usize;
-        let proof_task = ProofTaskManager::new(
+        let storage_worker_count = config.storage_worker_count();
+        let proof_task = match ProofTaskManager::new(
             self.executor.handle().clone(),
             state_root_config.consistent_view.clone(),
             task_ctx,
             max_proof_task_concurrency,
-        );
+            storage_worker_count,
+        ) {
+            Ok(task) => task,
+            Err(error) => {
+                // Fall back to parallel state root if proof task manager fails to initialize
+                tracing::error!(
+                    target: "engine::tree",
+                    ?error,
+                    "Failed to initialize proof task manager, falling back to parallel state root"
```

Comment:
super duper nit: at this point the payload processor only knows that it failed to initialize, it doesn't know what the consequences of that will be (ie that it will fallback to parallel state root). I would move this error message outside of this function and into the caller

Reply:
yeah that makes sense, since we are alraedy propagting it up and only the parent knows

Reply:
this makes sense, given that only the parent knows. and its already propogated